src/client/partition.rs

```diff
@@ -97,6 +97,9 @@ pub enum OffsetAt {
 
     /// The latest existing record.
     Latest,
+
+    /// Timestamp
```

Comment:
You may want move the warning about potentially broken behavior from the enum-level (see "Timstamp-based Queries") to this variant.